src/javascript-wrapper.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * File: javascript-wrapper.js 
+ */
+var _ = require('underscore');
+
+var createLm = require('./create-lm');
+var createState = require('./create-state');
+var factory = require('./pipeline-component-factory');
+var util = require('util');
+
+/**
+ * RDF Pipeline javascript wrapper fRunUpdater API as documented here: 
+ *    https://github.com/rdf-pipeline/noflo-rdf-pipeline/wiki/Wrapper-API
+ * 
+ * @param updater component updater function
+ * @param updaterArgs the introspected arguments found for the component updater.
+ *        used to extract the input for the updater
+ * @param vni a virtual node instance
+ */
+var fRunUpdater = function( updater, updaterArgs, vni ) {
+
+    // Default to no updater parameters; if no arguments, we won't go through the map
+    var updaterParameters = [];
+
+    // Have one or more updater parameters - get them in the right order and pass them 
+    updaterParameters = _.map( updaterArgs, function( arg ) { 
+       var state = vni.inputStates( arg );
+       if ( _.isUndefined( state ) ) { 
+           return undefined;
+       } else if (_.isArray( state )) { 
+           // Have a parameter associated with a port with multiple inputs ->
+           //  get an array of the data elements
+           return _.pluck(state, 'data');
+       } else if ( _.isObject( state ) ) {
+           // Just one input on this port parameter - get it
+           return state.data;
+       } else {
+           throw Error( "FRunupdater found an unexpected state: ", state );
+       }
+    });
+    
+    // Execute the updater on the VNI context, passing the updater Parameters as the API arguments
+    return new Promise( function(resolve ) { 
+
+         var results = updater.apply( vni, updaterParameters );
+         if ( results !== null ) { 
+             // If the updater returned anything, set the output state with it
+             vni.outputState( createState( vni.vnid, results ) );
+         } 
+         resolve( results );
+
+    }).catch( function( e ) { 
+         var outputState = vni.outputState();
+         if ( _.isUndefined( outputState.error ) || ! outputState.error  ) { 
+             outputState.error = true;
+             outputState.data = e.toString();
+             outputState.lm = createLm();
+             vni.outputState( outputState );
+         } 
+    });
+
+};
+
+/**
+ * This module provides a Javascript wrapper for updaters.  It is responsible for setting up the
+ * code to ensure updater callback, and then invoking the RDF pipeline factory to complete building
+ * the component and setting up the vni and state metadata.
+ *
+ * @param nodeDefOrUpdater 
+ *
+ * @this the context for this wrapper is the node instance.
```

Comment:
AFAICT @this is not really used.  It is passed to fRunUpdater on line 137, but fRunUpdater does not use it.  I suggest deleting the @this comment on line 71 and changing line 137 to use _.partial instead of .bind .